events/events.go

```diff
@@ -17,112 +17,125 @@ import (
+		return fmt.Errorf("error initiating db tx: %w", tx.Error)
+	}
 
-		selectEventsQuery := `
-			SELECT * FROM event_queue
-			WHERE
-				attempts <= 3 OR ((now() - last_attempt) > '1 hour'::interval)
-			FOR UPDATE SKIP LOCKED
-			LIMIT 1
-		`
-		err := tx.Raw(selectEventsQuery).First(&event).Error
-		if err != nil {
-			// Commit the transaction in case of no records found to prevent
-			// creating dangling connections and to release the locks
-			tx.Commit()
-			return err
-		}
+	selectEventsQuery := fmt.Sprintf(`
+		SELECT * FROM event_queue
+		WHERE attempts <= %d OR ((now() - last_attempt) > '1 hour'::interval)
+		FOR UPDATE SKIP LOCKED
+		LIMIT %d
+	`, eventMaxAttempts, eventQueueBatchSize)
 
-		switch event.Name {
-		case EventResponderCreate:
-			err = reconcileResponderEvent(tx, event)
-		case EventCommentCreate:
-			err = reconcileCommentEvent(tx, event)
-		default:
-			logger.Errorf("Invalid event name: %s", event.Name)
-			return tx.Commit().Error
-		}
+	var event api.Event
+	err := tx.Raw(selectEventsQuery).First(&event).Error
+	if err != nil {
+		// Commit the transaction in case of no records found to prevent
+		// creating dangling connections and to release the locks
+		tx.Commit()
+		return err
+	}
 
-		if err != nil {
-			errorMsg := err.Error()
-			setErr := tx.Exec("UPDATE event_queue SET error = ?, attempts = attempts + 1 WHERE id = ?", errorMsg, event.ID).Error
-			if setErr != nil {
-				logger.Errorf("Error updating table:event_queue with id:%s and error:%s. %v", event.ID, errorMsg, setErr)
-			}
-			return tx.Commit().Error
+	switch event.Name {
+	case EventResponderCreate:
+		err = reconcileResponderEvent(tx, event)
+	case EventCommentCreate:
+		err = reconcileCommentEvent(tx, event)
+	case EventPushQueueCreate:
+		if config.UpstreamConf.IsFilled() {
+			upstreamPushEventHandler := newPushToUpstreamEventHandler(config.UpstreamConf)
+			err = upstreamPushEventHandler.Run(ctx, tx, []api.Event{event})
 		}
+	default:
+		logger.Errorf("unrecognized event name: %s", event.Name)
+		return tx.Commit().Error
+	}
 
-		err = tx.Delete(&event).Error
-		if err != nil {
-			logger.Errorf("Error deleting event from event_queue table with id:%s", event.ID.String())
-			return tx.Rollback().Error
+	if err != nil {
+		errorMsg := err.Error()
+		setErr := tx.Exec("UPDATE event_queue SET error = ?, attempts = attempts + 1 WHERE id = ?", errorMsg, event.ID).Error
+		if setErr != nil {
+			logger.Errorf("error updating table:event_queue with id:%s and error:%s. %v", event.ID, errorMsg, setErr)
 		}
 		return tx.Commit().Error
+	}
 
+	err = tx.Delete(&event).Error
+	if err != nil {
+		logger.Errorf("error deleting event from event_queue table with id:%s", event.ID.String())
+		return tx.Rollback().Error
 	}
+	return tx.Commit().Error
+}
 
-	// Keep on iterating till the queue is empty
+// consumeEventsUntilEmpty consumes events forever until the event queue is empty.
+func consumeEventsUntilEmpty(ctx context.Context, config Config) {
```

Comment:
The name is misleading, it should indicate that it keeps on polling events in  an infinite for loop

Reply:
I think it's fairly descriptive - the loop is not really infinite because it terminates when the queue is empty.  And it's not necessary to mention how the queue is fully consumed; whether it's with infinite for loop or something else.

Do you have any suggestion?

Reply:
Sorry my bad, I confused it with the Listen events func